src/connection.cr

```diff
@@ -21,16 +21,31 @@ module Redis
     # SSL connections require specifying the `rediss://` scheme.
     # Password authentication uses the URI password.
     # DB selection uses the URI path.
+    #
+    # To use Unix sockets: `socket:/path/to/redis.sock`
     def initialize(@uri = URI.parse("redis:///"))
-      host = uri.host || "localhost"
-      port = uri.port || 6379
-      socket = TCPSocket.new(host, port)
-      socket.sync = false
-
-      # Check whether we should use SSL
-      if uri.scheme == "rediss"
-        socket = OpenSSL::SSL::Socket::Client.new(socket)
+      if uri.scheme == "socket"
+        socket = UNIXSocket.new(uri.path)
+        socket.sync = false
+        db = "0"
```

Comment:
I see why you did it this way, but it seems like it would be impossible to use a DB slot other than 0 while using a Unix socket.

Reply:
Have to admit I only learned later what the DB slots are. But agreed that it does not make sense to exclude the functionality depending on the connection type without any needs.

I'll change it when I have some time.

Reply:
Following along the same lines as the other reply, if other Redis clients have opinions on how to select a different DB with a Unix socket URL, I'm happy to follow suit here.